src/crimson/os/seastore/onode_manager/staged-fltree/node.cc

```diff
@@ -1828,6 +1829,7 @@ LeafNode::erase(context_t c, const search_position_t& pos, bool get_next)
     }
     return seastar::now().then(
         [c, &pos, this_ref = std::move(this_ref), this, FNAME] () mutable {
+      assert_moveable(this_ref);
```

Comment:
I see.  without mutable, captured this_ref will be constant and move will call copy constructor instead of move constructor, which will cause refence++. This assert is to guarantee mutable will not be removed by accident.

Reply:
Exactly.

Reply:
seems better to put this assert in every this_ref removed capture body.